src/Test/Perf/infra/fetch_build.csx

```diff
@@ -4,39 +4,81 @@
 
-var sourceFolder = $@"\\cpvsbuild\drops\Roslyn\Roslyn-{branch}-Signed-Release";
+// Get the last successful build that was tested
+string lastSuccessfulBuild = null;
+var lastSuccessfulDirectory = Environment.ExpandEnvironmentVariables($@"%SYSTEMDRIVE%\Last-Successful-{Path.GetFileName(sourceFolder)}");
+var buildInfoPath = Path.Combine((lastSuccessfulDirectory), "BuildNumber.txt");
 
-string latestBuild = null;
-foreach (var folder in Directory.GetDirectories(sourceFolder, "????????.?").Reverse())
+if (Directory.Exists(lastSuccessfulDirectory) && File.Exists(buildInfoPath))
 {
-    if (SanityTestPassesForBuild(folder))
-    {
-        latestBuild = folder;
-        break;
-    }
+    lastSuccessfulBuild = File.ReadAllText(buildInfoPath);
 }
 
-if (latestBuild == null)
+// Start monitoring the share for new successful builds
+Task.Run(() => StartWatching(sourceFolder, destinationFolder, lastSuccessfulBuild, buildInfoPath));
+Console.WriteLine("Press any key to continue");
+Console.Read();
+
+void StartWatching(
+    string sourceFolder,
+    string destinationFolder,
+    string lastSuccessfulBuild,
+    string buildInfoPath)
 {
-    throw new InvalidOperationException($"Could not locate build with passing tests at location \"{sourceFolder}\".");
-}
+    string latestBuild = null;
+    var lastSuccessfulBuildNumber = lastSuccessfulBuild == null ? 0 : Convert.ToInt32(Convert.ToDouble(lastSuccessfulBuild) * 100);
+    while (true)
+    {
+        foreach (var folder in Directory.GetDirectories(sourceFolder, "????????.?").Reverse())
+        {
+            var buildDirectoryName = Path.GetFileName(folder);
+            var buildDirectoryNumber = Convert.ToInt32(Convert.ToDouble(buildDirectoryName) * 100);
 
-var latestBuildFolder = Path.Combine(sourceFolder, latestBuild);
+            // if we are no longer looking at the newer builds, then quit searching
+            if (!(buildDirectoryNumber > lastSuccessfulBuildNumber))
+            {
+                break;
+            }
 
-if (Directory.Exists(destinationFolder))
-{
-    Directory.Delete(destinationFolder, recursive: true);
-}
+            if (SanityTestPassesForBuild(folder))
+            {
+                // Copy the binaries from the drop to Release binaries
+                latestBuild = folder;
+                var latestBuildFolder = Path.Combine(sourceFolder, latestBuild);
 
-CopyDirectory(latestBuildFolder, destinationFolder);
+                if (Directory.Exists(destinationFolder))
+                {
+                    Directory.Delete(destinationFolder, recursive: true);
+                }
+
+                CopyDirectory(latestBuildFolder, destinationFolder);
+
+                // Start Automation
+                // Fire and forget
+                var processInfo = new ProcessStartInfo(@"TriggerAutomation.bat");
+                processInfo.Arguments = $"{buildInfoPath} {buildDirectoryName} {sourceFolder}";
+                var process = Process.Start(processInfo);
```

Comment:
Flow is `MonitorShareTask` calls `fetch_build`, which calls `TriggerAutomation` and kills itself. This way, there is no other active running processes in the machine. Once the tests are run, we start `MonitorShareTask` again. Maybe instead of running `MonitorShareTask` at the end, I can run the `fetch_build` directly